tests/console/docker_image.pm

```diff
@@ -0,0 +1,63 @@
+# SUSE's openQA tests
+#
+# Copyright © 2018 SUSE LINUX GmbH
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Test installation and running of the docker image for this snapshot
+# Maintainer: Fabian Vogt <fvogt@suse.com>
+
+use base 'consoletest';
+use testapi;
+use utils;
+use strict;
+
+sub run {
+    select_console 'root-console';
+
+    check_var('VERSION', 'Tumbleweed') || die 'Only Tumbleweed uses this test';
+
+    my $rpm_name   = 'opensuse-image';
+    my $image_name = 'opensuse:tumbleweed';
+
+    # Install the image
+    zypper_call "in docker $rpm_name";
+
+    # Start the docker daemon, normally done by previous test modules already
+    systemctl 'start docker';
+    systemctl 'status docker';
+    assert_script_run 'docker info';
+
+    # Load the image
+    assert_script_run 'docker load -i /usr/share/suse-docker-images/native/*-image*.tar.xz';
+
+    # Show that the image got registered
+    assert_script_run "docker images $image_name";
+
+    # Running executables works
+    assert_script_run qq{docker container run --rm $image_name echo "I work" | grep "I work"};
+
+    # It is openSUSE Tumbleweed
+    assert_script_run qq{docker container run --rm $image_name cat /etc/os-release | grep 'PRETTY_NAME="openSUSE Tumbleweed"'};
+
+    # Zypper and network are working
+    assert_script_run qq{docker container run --rm $image_name zypper -v ref | grep "All repositories have been refreshed"};
+
+    # Interactive session works
+    type_string <<"EOF";
+docker container run --rm -it $image_name /bin/bash; echo DOCKER-\$?- > /dev/$serialdev
+exit 42
+EOF
+
```

Comment:
If you like instead of using all these file comments you can also specify a fail_message, e.g.

```perl
assert_script_run 'docker load -i /usr/share/suse-docker-images/native/*-image*.tar.xz', fail_message => 'Image could not be loaded';
assert_script_run "docker images $image_name", fail_message => 'Image was not registered';
assert_script_run qq{docker container run --rm $image_name echo "I work" | grep "I work"}, fail_message => 'Could not run executables';
assert_script_run qq{docker container run --rm $image_name grep 'PRETTY_NAME="openSUSE Tumbleweed"' /etc/os-release}, fail_message => 'Could not find openSUSE Tumbleweed';
assert_script_run qq{docker container run --rm $image_name zypper -v ref | grep "All repositories have been refreshed"}, fail_message => 'zypper not working';
```

Reply:
That loses the clean indentation though :-/

Reply:
you mean the additional blank lines? I consider this a bit too verbose

Reply:
No, that each comment/explanation starts at the same line offset. I can remove the blanks.